signer/gpg2/gpg2_test.go

```diff
@@ -529,6 +537,8 @@ func TestGPG2Signer_SignFiles(t *testing.T) {
 		})
 	}
 
+	// This is not an ideal check, and can cause failures when other code
+	// or tests fail to clean up properly - but it's probably good enough.
 	matches, err := filepath.Glob("/tmp/autograph_*sign_files*")
```

Comment:
Ugh, I, unfortunately, think this is something that will definitely cause flakes. Would it be possible to pass in a `prefix string` into `gpg2.New` and use it everywhere we're creating temp files? The place its called in `addSigners` could just us a static prefix (instead of being configurable) for now, but it would clear up all the other problems that this could happen might be straight-forward if we do that. 

(I could be wrong about it being straight-forward and I'm happy to hear I've missed something important. I know it would mean having to look at all the other `os.Remove*` deals.)

Reply:
Hm, I'm not sure that will guarantee that we're free of flakes still, but it wouldn't certtainly reduce the possibility. Maybe we ought to pass a directory to create the keyring in instead, as I'm already doing in the other test in this patch? That way there's no globbing or guesswork that has potential interference.

Reply:
So sorry, yes, I def meant what you're describing and didn't think through the mechanics of needing to pass in a directory vs a prefix passed in when creating that directory (and then deleteling stuff that only matched the temp dir with that prefix; that's where I failed to think it through: the prefix would only be in the middle of the path!)

Reply:
I think this is all fixed now. Thanks for the push on it - I ended up catching a longstanding issue with this assertion because of it!